spec/unit/remotes_spec.rb

```diff
@@ -1,39 +1,159 @@
 require '3scale_toolbox'
 
 RSpec.describe ThreeScaleToolbox::Remotes do
-  let(:test_class) do
-    Class.new { include ThreeScaleToolbox::Remotes }
+  include_context :resources
+  include_context :temp_dir
+
+  RSpec.shared_examples 'raises invalid url error' do
+    it 'raises error' do
+      expect { subject.add_uri('some_name', origin) }.to raise_error(ThreeScaleToolbox::InvalidUrlError,
+                                                                     /invalid url/)
+    end
   end
 
-  subject { test_class.new }
+  let(:yaml_content) do
+    <<~HEREDOC
```

Comment:
if you use `YAML` instead of `HEREDOC` then some editors will be able to actually do syntax highlighting on the string.